cpp/GRPCTest.cpp

```diff
@@ -175,40 +179,49 @@ LUA_FUNCTION(UpdateDiscordStatus) {
     LUA->GetField(1, "instance");
     discordP.instance = LUA->GetNumber();
 
+    int buttonI = 0;
+
     LUA->GetField(1, "buttonPrimaryLabel");
-    if (LUA->GetType(-1) == GarrysMod::Lua::Type::String) {
-        DiscordRichPresenceButton button;
+    if (LUA->GetString() != NULL) {
+        discordP.buttons[buttonI] = (DiscordRichPresenceButton *)malloc(
+            sizeof(DiscordRichPresenceButton));
 
-        button.label = LUA->GetString();
+        discordP.buttons[buttonI]->label = LUA->GetString();
 
         LUA->GetField(1, "buttonPrimaryUrl");
         LUA->CheckString();
 
-        button.url = LUA->GetString();
+        discordP.buttons[buttonI]->url = LUA->GetString();
 
-        discordP.buttons[0] = &button;
+        buttonI += 1;
     }
 
     LUA->GetField(1, "buttonSecondaryLabel");
-    if (LUA->GetType(-1) == GarrysMod::Lua::Type::String) {
-        DiscordRichPresenceButton button;
+    if (LUA->GetString() != NULL) {
+        discordP.buttons[buttonI] = (DiscordRichPresenceButton *)malloc(
+            sizeof(DiscordRichPresenceButton));
 
-        button.label = LUA->GetString();
+        discordP.buttons[buttonI]->label = LUA->GetString();
 
         LUA->GetField(1, "buttonSecondaryUrl");
         LUA->CheckString();
 
-        button.url = LUA->GetString();
-
-#pragma warning(disable : 6201)
-        discordP.buttons[1] = &button;
+        discordP.buttons[buttonI]->url = LUA->GetString();
     }
 
     Discord_UpdatePresence(&discordP);
     return 0;
 }
 
 GMOD_MODULE_OPEN() {
+    // Set the current version as a vector
+    std::vector<int> version = {DISCORD_RPC_VERSION};
+
+    Vector semver;
+    semver.x = version[0];
+    semver.y = version[1];
+    semver.z = version[2];
+
```

Comment:
I'm not sure how I feel about this being a vector - it doesn't feel intuitive to me to do something like `DiscordRPCVersion.y == 2`

How would you feel about changing this into a three-digit integer e.g. DiscordRPCVersion = 121. You could then do some semver checks like `if DiscordRPCVersion >= 120 and DiscordRPCVersion < 130 then` without worrying about a vector?

It'd also make the C++ code a bit simpler since we don't need to import vector.

I'm assuming you've implemented this for some changes you've got in a fork so I'd like to know your thoughts and what your usecase for this one is!

Reply:
It is a common practice to use vectors for semver as you can compare the version just by `DiscordRPCVersion == Vector(1, 2, 1)`. However, there are also versioning with numerical values just like you mentioned. So I say it's just a choice of design.

I think it is also possible to use Vector versioning without including `stdlib.h`, it is by just using the one in the GLua header files and avoiding using `std::vector`.

If the numerical approach is more appropriate in your eyes, I'm creating another commit to change it to number-based from vector-based